drop-transfer/src/service.rs

```diff
@@ -111,6 +114,95 @@ impl Service {
         &self.state.storage
     }
 
+    fn trigger_peer_incoming(&mut self, xfer: Arc<crate::IncomingTransfer>) {
+        debug!(self.logger, "trigger_peer_incoming() called: {:?}", xfer);
+
+        let id = xfer.id();
+
+        if let Some(handle) = self.tasks.get(&id) {
+            if !handle.is_finished() {
+                return;
+            } else {
+                self.tasks.remove(&id);
+            }
```

Comment:
```suggestion
            }
```
It's a minor thing but the `insert()` below will remove it anyway in a more efficient way

Reply:
updated it